libnmstate/nm/checkpoint.py

```diff
@@ -96,16 +99,24 @@ def create(self):
             CHECKPOINT_CREATE_FLAG_DELETE_NEW_CONNECTIONS |
             CHECKPOINT_CREATE_FLAG_DISCONNECT_NEW_DEVICES
         )
-        cp = self._manager.interface.CheckpointCreate(devs, timeout, cp_flags)
-        logging.debug('Checkpoint %s created for all devices: %s', cp, timeout)
-        self._checkpoint = cp
+        try:
+            dbuspath = self._manager.interface.CheckpointCreate(devs, timeout,
+                                                                cp_flags)
+            logging.debug('Checkpoint %s created for all devices: %s',
+                          dbuspath, timeout)
+            self._checkpoint = dbuspath
+        except dbus.exceptions.DBusException:
+            pass
```

Comment:
If `create` fails, it should explode.
If the caller is interested in doing something when it explodes, he should handle it.

If there is a need to ignore an exception, a very good reason for it is needed, otherwise it is a smell of something bad happening.

All the `if` checks below I consider a bad practice. If a caller uses them on an object that is not properly initialized, it should explode as well.

Reply:
Can you please explain how to envision this with a context manager? Do you want something like this?

```
try:
    with nm.checkpoint.CheckPoint() as checkpoint:
        with _setup_providers():
            _add_interfaces(desired_state.interfaces, current_state.interfaces)
        with _setup_providers():
            current_state = state.State(
                {Constants.INTERFACES: netinfo.interfaces()}
            )
            _edit_interfaces(desired_state, current_state)
        if verify_change:
            _verify_change(desired_state)
except NMCheckPointError:
    raise NmstateConflictError('Error creating a check point')
```

Reply:
Looks good to me.
Consider moving the try block into its own function so this will look a bit nicer.

Reply:
Changed this ... did not move this in its own function because this was already a rebase hell and it is most of what the function already does. It might make sense to clean this up once this becomes a method of a state object so that there would not be a ton of variables to pass around.